src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -239,6 +265,41 @@ public static Duration parseDuration(String duration) throws ParseException {
         }
     }
 
+    /**
+     * A helper function to check if the hours and minutes satisfies duration constraints.
+     *
+     * @throws ParseException if the {@code hours} and {@code minutes} combined exceeds 336 hours (2 weeks)
+     */
+    private static void checkValidHoursAndMinutes(int hours, int minutes) throws ParseException {
+        checkValidMinutes(minutes);
+        checkValidHours(hours);
+        if (hours == 336 && minutes > 0) {
+            throw new ParseException("The event's duration cannot exceed 2 weeks! (336 hours)");
+        }
+    }
+
+    /**
+     * A helper function to check if the hours satisfy duration constraints.
+     *
+     * @throws ParseException if the given {@code hours} is less than 0 or more than 336
+     */
+    private static void checkValidHours(int hours) throws ParseException {
+        if (hours < 0 || hours > 336) {
+            throw new ParseException("The event's duration cannot exceed 2 weeks! (336 hours)");
+        }
+    }
+
+    /**
+     * A helper function to check if the minutes satisfy minute constraints.
+     *
+     * @throws ParseException if the given {@code minutes} is less than 0 or larger than 59
+     */
+    private static void checkValidMinutes(int minutes) throws ParseException {
+        if (minutes < 0 || minutes > 59) {
+            throw new ParseException("Minutes should be an integer between 0 to 59 inclusive!");
+        }
+    }
+
```

Comment:
Is there a reason to have this? Or it's just something we do to limit the duration on our side? What about it's a event with recurrence that lasts a month?

Reply:
We are just limiting the duration on our side, otherwise it allows users to key in arbitrarily large durations which could potentially cause bugs. We can change the exact limit, is there a specific number you would recommend?